cli/opcodes_dictionary.rs

```diff
@@ -0,0 +1,222 @@
+pub struct OpCodeDescription {
+    pub name: &'static str,
+    pub description: &'static str,
+}
+
+pub const OPCODE_DESCRIPTIONS: [OpCodeDescription; 189] = [
+OpCodeDescription { name: "Abortable", description: "Verify that an Abort can happen. Assert if an Abort at this point might cause database corruption. This opcode only appears in debugging builds. An Abort is safe if either there have been no writes, or if there is an active statement journal." },
+OpCodeDescription { name: "Add", description: "Add the value in register P1 to the value in register P2 and store the result in register P3. If either input is NULL, the result is NULL." },
+OpCodeDescription { name: "AddImm", description: "Add the constant P2 to the value in register P1. The result is always an integer. To force any register to be an integer, just add 0." },
+OpCodeDescription { name: "Affinity", description: "Apply affinities to a range of P2 registers starting with P1. P4 is a string that is P2 characters long. The N-th character of the string indicates the column affinity that should be used for the N-th memory cell in the range." },
```

Comment:
would it be better if we use a hashmap instead of a dictionary?

then i think will not require for loop?

please, correct me if i am wrong

Reply:
Yes. It would be faster to find a specific opcode with a HashMap. Nevertheless, this is a vector of ~200 items which isn't big at all and... `.opcodes` is a command you don't care about to over optimize.